doc/sources/algorithms.rst

```diff
@@ -20,177 +20,331 @@
+   * - `Lasso`
+     - All parameters are supported except:
+
+       - ``sample_weight`` != `None`
+     - Multi-output and sparse data are not supported, `#observations` should be >= `#features`.
+
+Clustering
+**********
+
+.. list-table::
+   :widths: 10 30 20
+   :header-rows: 1
+   :align: left
+
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `KMeans`
+     - All parameters are supported except:
+
+       - ``precompute_distances``
+       - ``sample_weight`` != `None`
+     - No limitations
+   * - `DBSCAN`
+     - All parameters are supported except:
+
+       - ``metric`` != `'euclidean'` or `'minkowski'` with ``p`` != `2`
+       - ``algorithm`` != `'brute'` or `'auto'`
+     - Only dense data is supported
+
+Dimensionality reduction
+************************
+
+.. list-table::
+   :widths: 10 30 20
+   :header-rows: 1
+   :align: left
+
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `PCA`
+     - All parameters are supported except:
+
+       - ``svd_solver`` != `'full'`
+     - Sparse data is not supported
+   * - `TSNE`
+     - All parameters are supported except:
+
+       - ``metric`` != 'euclidean' or `'minkowski'` with ``p`` != `2`
+
+       Refer to :ref:`TSNE acceleration details <acceleration_tsne>` to learn more.
+     - Sparse data is not supported
+
+Other tasks
+***********
+
+.. list-table::
+   :widths: 10 30 20
+   :header-rows: 1
+   :align: left
+
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `NearestNeighbors`
      - 
        - For ``algorithm`` == 'kd_tree':
          
-         all parameters except ``metric`` != 'euclidean' or 'minkowski' with ``p`` != 2
+         all parameters except ``metric`` != `'euclidean'` or `'minkowski'` with ``p`` != `2`
        - For ``algorithm`` == 'brute':
          
-         all parameters except ``metric`` not in ['euclidean', 'manhattan', 'minkowski', 'chebyshev', 'cosine']
-     - Sparse data is not supported.
-   * - Other
-     - train_test_split
-     - All parameters are supported.
-     - Only dense data is supported.
-   * - Other
-     - assert_all_finite
-     - All parameters are supported.
-     - Only dense data is supported.
-   * - Other
-     - pairwise_distance
-     - With ``metric`` = 'cosine' or 'correlation'.
-     - Only dense data is supported.
-   * - Other
-     - roc_auc_score
-     - Parameters ``average``, ``sample_weight``, ``max_fpr`` and ``multi_class`` are not supported.
-     - No limitations.
+         all parameters except ``metric`` not in [`'euclidean'`, `'manhattan'`, `'minkowski'`, `'chebyshev'`, `'cosine'`]
+     - Sparse data is not supported
+   * - `train_test_split`
+     - All parameters are supported
+     - Only dense data is supported
+   * - `assert_all_finite`
+     - All parameters are supported
+     - Only dense data is supported
+   * - `pairwise_distance`
+     - With ``metric`` = `'cosine'` or `'correlation'`.
```

Comment:
Can someone help me rephrase this in the same way as it's done for other algorithms?

Reply:
I suppose that @KalyanovD may help you

Reply:
"All parameters except metric != cosine and correlation"?

Reply:
I've decided to use "not in" phrase